include/zephyr/drivers/video.h

```diff
@@ -202,6 +202,33 @@ struct video_frmival_enum {
 	};
 };
 
+/**
+ * @brief Port / Endpoint DT Helpers
+ *
+ */
+
+/* Drivers should not use this except they know exactly that port has a pid */
+#define _DT_INST_PORT_BY_ID(n, pid)                                                                \
+	COND_CODE_1(DT_NODE_EXISTS(DT_INST_CHILD(n, ports)), (DT_CHILD(DT_INST_CHILD(n, ports), port_##pid)), (DT_INST_CHILD(n, port_##pid)))
```

Comment:
Sorry if I'm missing something here, why not use this pattern? Or remove the underscore if users should be using this macro directly?

Reply:
We can use this pattern directly if we are sure that the target port has an ID, e.g. port_0, port_1, etc. It will not work in case we have only 1 port with no ID. I will remove the underscore as this macro can be used both internally and publicly.